kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/position/MMLocationParser.java

```diff
@@ -0,0 +1,285 @@
+    }
+
+    public boolean isFixed() {
+        return this.isFix;
+    }
+
+    public NmeaPosition getNmeaPosition() {
+        return new NmeaPosition(this.lat, this.lon, this.alt, this.speed, this.track, this.fixQuality,
+                this.nrSatellites, this.mDOP, this.mPDOP, this.mHDOP, this.mVDOP, this.fixQuality);
+    }
+
+    public String getNmeaTime() {
+        return this.localTime.toString();
+    }
+
+    public String getNmeaDate() {
+        return this.localDate.toString();
+    }
+
+    public void parseRawLocation(Variant<?> rawLocationVariant) {
+        Map<String, Variant<?>> locationData = (Map<String, Variant<?>>) rawLocationVariant.getValue();
+        for (Map.Entry<String, Variant<?>> rawEntry : locationData.entrySet()) {
+
+            switch (rawEntry.getKey()) {
+            case "latitude":
+                this.lat = Math.toRadians((Double) rawEntry.getValue().getValue());
+                break;
+
+            case "longitude":
+                this.lon = Math.toRadians((Double) rawEntry.getValue().getValue());
+                break;
+
+            case "altitude":
+                this.alt = (Double) rawEntry.getValue().getValue();
+                break;
+
+            // time comes in format HHmmss.SS, so we cut the string after the dot to extract only the util information
+            case "utc-time":
+                String time = ((String) rawEntry.getValue().getValue()).toString();
+                this.localTime = LocalTime.parse(time.split("\\.")[0], DateTimeFormatter.ofPattern("HHmmss"));
+                break;
+
+            default:
+                // DO NOTHING
+            }
+        }
+    }
+
+    public void parseNmeaLocation(Variant<?> nmeaLocationVariant) {
+        try {
+            String locationString = ((CharSequence) nmeaLocationVariant.getValue()).toString();
+
+            List<String> nmeaSentences = Arrays.asList(locationString.split("\\r?\\n|\\r")).stream()
+                    .filter(sentence -> !sentence.isEmpty()).collect(Collectors.toList());
+
+            for (String sentence : nmeaSentences) {
+
+                int starpos = sentence.indexOf('*');
+                final List<String> tokens = Arrays.asList(sentence.substring(0, starpos).split(","));
+
+                String sentenceGnss = sentence.substring(1, 3);
+
+                this.gnssTypes.add(getGnssTypeFromSentenceId(sentenceGnss));
+
+                String sentenceType = sentence.substring(3, 6);
+
+                switch (sentenceType) {
+
+                case "GSA":
+                    parseGsaSentence(tokens);
+                    break;
+
+                case "RMC":
+                    parseRmcSentence(tokens);
+                    break;
+
+                case "GGA":
+                    parseGgaSentence(tokens);
+                    break;
+
+                default:
+                    // Do Nothing
+
+                }
+            }
+
+        } catch (Exception ex) {
```

Comment:
Here you catch all the exceptions, but the code above does not throw anything. Do you implemented this for catching the unchecked exceptions, like the `NumberFormatException`?